app.js

```diff
@@ -1,46 +1,138 @@
+                "name": fields[0].trim(), // remove trailing whitespaces
+                "score": fields[1].replace(/[^\d.]/g, "") // remove all except numbers and decimal points
             });
+            // check if the line contains a label for sloth with confidence higher than 70%
+            if(fields[0].includes("sloth") && !fields[0].includes("Ursus ursinus")) {
+                if(fields[1].replace(/[^\d.]/g, "") > 0.70) {
+                    sloth = true;
+                }
+            }
         }
     });
+    resultObject.sloth_check.push({
+        "contains_sloth": sloth
+    });
+    return resultObject;
 }
 
-// HTTP GET root path
-app.get("/", function (req, res) {
-    res.send("Hello world!");
-});
+/**
+ * Handles passing the base64 string to all proper methods, which includes checking its file type and writing the
+ * file to disk
+ * @param {string} base64String - base64 string of file
+ * @returns {*} - file path if the argument provided was PNG or JPEG, null otherwise
+ */
+function processImage(base64String) {
+    let fileType = checkFileType(base64String); // checking the file type of the base64
+    if (fileType.includes("PNG")) {
+        writePngToDisk(base64String);
+    } else if (fileType.includes("JPEG")) {
+        writeJpegToDisk(base64String);
+    } else {
+        return null; // return null if the file type isn't PNG or JPEG
+    }
+    return "../slothbucket/out.jpeg"; // return file path if the file type is PNG or JPEG
+}
 
-// HTTP get test path (temporary path for manual testing)
-app.get("/test", function (req, res) {
-    child_process.execFile("python", ["classify_image.py"], {
-        cwd: "../root"
-    }, function (error, result) {
-        parseTensorFlowResult(result);
-        res.send(result.toString().split("\n"));
-    });
+// HTTP POST for any TensorFlow image recognition with bonus sloth check
+app.post("/sloth-check", function (req, res) {
+    if(req.body.base64 !== null && req.body.base64 !== undefined) {
+        let filePath = processImage(req.body.base64); // sending to this function to check file type and save to disk
+        if (filePath !== null) {
+            // commands to run TensorFlow
+            child_process.execFile("python", ["classify_image.py", "--image_file", filePath], {
+                cwd: "../root" // change working directory
+            }, function (error, result) {
+                if (error !== null) {
+                    console.log(error);
+                }
+                res.status(200).send(parseTensorFlowResult(result)); // success
+            });
+        } else {
+            // if filePath is null, that means the image wasn't able to be processed (e.g., it wasn't a JPEG or PNG)
+            res.status(500).send({status:500, message:"File type not supported"});
+        }
+    }
+    else {
+        // body was undefined or null
+        res.status(500).send({status:500, message:"Cannot read undefined body"});
+    }
```

Comment:
This logic as well as the associated functions might be better fit being put into a module that gets required in `app.js`